metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -1445,6 +1445,51 @@ def run_assertions_for_asset(
 
         return res["runAssertionsForAsset"]
 
+    def get_entities_v2(
+        self,
+        entity_name: str,
+        urns: List[str],
+        aspects: List[Type[Aspect]],
+        with_system_metadata: bool = True,
+    ) -> Optional[Dict]:
+        payload = {
+            "urns": urns,
+            "aspectNames": list(map(lambda x: x.ASPECT_NAME, aspects)),
+            "withSystemMetadata": with_system_metadata,
+        }
+        headers: Dict[str, Any] = {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+        url = f"{self.config.server}/openapi/v2/entity/batch/{entity_name}"
+        try:
+            response = self._session.post(
+                url, data=json.dumps(payload), headers=headers
+            )
+            if response.status_code != 200:
+                logger.debug(
+                    f"Non 200 status found while fetching entities - {response.status_code}"
+                )
+                return None
+            json_resp = response.json()
+            entities = json_resp.get("entities", [])
+            retval: Dict[str, Any] = {}
+            for entity in entities:
+                for aspect in aspects:
+                    entity_aspects = entity.get("aspects", {})
+                    entity_urn = entity.get("urn", None)
+                    if entity_urn is not None and aspect.ASPECT_NAME in entity_aspects:
+                        aspect_raw_value = entity_aspects.get(aspect.ASPECT_NAME).get(
+                            "value"
+                        )
+                        aspect_value = aspect.from_obj(aspect_raw_value)
```

Comment:
does this always work? I thought there's some differences between the openapi format and the restli format that we currently use?

note that from_obj ignores unknown fields, so it's possible this is silently dropping data - needs to be tested more completely

Reply:
What is the alternative though? We can return raw dict if that's preferred, but if the field is unknown to the model, what else can we do?

Reply:
I guess the alternative might be to raise if the dict has unknown fields? That means that this API becomes unusable for everyone every time a new field is introduced but python model is not up to date, whereas in the current implementation it's unusable only for those who depend on specific new field(s) that have to be added to the model anyway. In such case I'd vote for the current behavior.

Reply:
Also as a compromised, we can return both the raw dict and the native objects.

Reply:
My point was simply that from_obj is going to crash for a bunch of object types, because the formatting openapi uses is slightly different from the format used by restli. The python classes only support restli format and avro.

As a simple example, try this on dev01

```
from pprint import pp

import datahub.metadata.schema_classes as models
from datahub.ingestion.graph.client import get_default_graph

graph = get_default_graph()

urn = "urn:li:assertion:81c2e13a-1f41-43d5-8637-1171977badee"
info = graph.get_entities_v2(
    entity_name="assertion",
    urns=[urn],
    aspects=[models.AssertionInfoClass],
)
assert info is not None
pp(info[urn])
```

Reply:
For posterity: we discussed offline and agreed that at this point this method should return raw dict with aspects and it's up to the user to deserialize it as appropriate. This approach seems the most reliable and is not affected by silent data loss issue. As soon as data model consistency situation improves, we can implement proper deserialization support, perhaps as a wrapper method.